pandas/tests/io/formats/test_format.py

```diff
@@ -171,6 +171,15 @@ def test_repr_truncation(self):
         with option_context("display.max_colwidth", max_len + 2):
             assert "..." not in repr(df)
 
+    def test_repr_truncation_preserves_na(self):
+        # https://github.com/pandas-dev/pandas/issues/55630
+        with option_context("display.max_rows", 10):
```

Comment:
If you set this to 2 instead of 10, I think we could then test for the entire string.